torcheeg/trainers/classifier.py

```diff
@@ -266,4 +296,349 @@ def predict_step(self,
+        self.reset_metric()
+        
+    def validation_step(self, batch: Tuple[torch.Tensor],
+                        batch_idx: int) -> torch.Tensor:
+        x, y = batch
+        # calculate feat y_pred
+        feat = self.decoder(x)
+        y_hat = self.Fpredictor(feat)
+
+        # get loss (pred_loss,center loss,total_loss)
+        pre_loss = self.ce_fn(y_hat, y)
+        centerloss = self.Closs(y, feat)
+        loss = pre_loss + self.lammda * centerloss
+
+        # update  metrics
+        self.val_loss.update(loss)
+        self.center_loss_val.update(centerloss)
+        self.predict_loss_val.update(pre_loss)
+        self.val_metrics.update(y_hat, y)
+
+        # log loss
+        log_dict = {"Val Total Loss" : self.val_loss.compute(),
+                    "Val CELoss" : self.predict_loss_val.compute(),
+                    "Val CLoss": self.center_loss_val.compute() }
+        # log metrics
+        for i, metric_value in enumerate(self.val_metrics.values()):
+            log_dict[f"Val {self.metrics[i]}"] =  metric_value.compute()
+            self.log_dict(log_dict,
+                          prog_bar=True,
+                          on_epoch=True,
+                          on_step=False,
+                          logger=True)
+        
+        # print the metrics
+        
+        return loss
+        
+
+    def on_validation_epoch_end(self) -> None:
+        # log loss
+        log_dict = {"Val Total Loss" : self.val_loss.compute(),
+                    "Val CELoss" : self.predict_loss_val.compute(),
+                    "Val CLoss": self.center_loss_val.compute() }
+        # log classfication metrics
+        for i, metric_value in enumerate(self.val_metrics.values()):
+            log_dict[f"Val {self.metrics[i]}"] = metric_value.compute()
+        self.log_dict(log_dict,
+                    prog_bar=True,
+                    on_epoch=True,
+                     on_step=False,
+                     logger=True)
+        # reset the metrics
+        self.reset_metric("val")
+        str = "\n[Val] "
+        for key, value in self.trainer.logged_metrics.items():
+            if key.startswith("Val"):
+                str += f"{key}: {value:.3f} "
+        print(str + '\n')
+    
+
+
+    def test_step(self, batch: Tuple[torch.Tensor],
+                  batch_idx: int) -> torch.Tensor:
+        x, y = batch
+        # centerloss
+        feat = self.decoder(x) 
+        centerloss = self.Closs(y,feat)
+
+        # predict loss
+        y_hat = self.Fpredictor(feat)
+        pre_loss = self.ce_fn(y_hat, y)
+    
+        #Total loss 
+        loss = pre_loss +self.lammda *centerloss
+
+        self.test_loss.update(loss)
+        self.center_loss_test.update(centerloss)
+        self.predict_loss_test(pre_loss)
+        self.test_metrics.update(y_hat, y)
+        return loss
+    
+    def on_test_epoch_end(self) -> None:
+        log_dict = {"Test Total Loss" : self.test_loss.compute(),
```

Comment:
The logs are excessively long and the log messages are inconsistent. Please refer to the code below for the necessary modifications.

```
def on_test_epoch_end(self) -> None:
        self.log("test_loss", # just a test_loss, not Test Total Loss or something like that
                 self.test_loss.compute(),
                 prog_bar=False,
                 on_epoch=True,
                 on_step=False,
                 logger=True)
        for i, metric_value in enumerate(self.test_metrics.values()):
            self.log(f"test_{self.metrics[i]}",
                     metric_value.compute(),
                     prog_bar=False,
                     on_epoch=True,
                     on_step=False,
                     logger=True)

        # print the metrics
        str = "\n[Test] "
        for key, value in self.trainer.logged_metrics.items():
            if key.startswith("test_"):
                str += f"{key}: {value:.3f} "
        print(str + '\n')

        self.test_loss.reset()
        self.test_metrics.reset()
```

Reply:
Do you mean not to log center loss and cross_entropy loss？  log loss（sum of center loss and cross_entropy loss,correct? or just log cross entropy loss?

Reply:
Total loss, i.e., the sum of the center loss and the cross entropy loss.